src/util/bitvector_expr.h

```diff
@@ -778,6 +794,66 @@ inline binary_overflow_exprt &to_binary_overflow_expr(exprt &expr)
   return ret;
 }
 
+class plus_overflow_exprt : public binary_overflow_exprt
```

Comment:
Shouldn't these classes be used somewhere in the existing code?

Reply:
I had previously taken a look and it seems we consistently take shortcuts by creating `binary_overflow_exprt`s with a computed id.

Reply:
I have now pushed additional commits which use the new classes throughout the existing code where the IDs were previously being checked.